src/lib.rs

```diff
@@ -86,53 +88,36 @@ pub fn db_conn_string(config: &Config) -> Result<String> {
         .collect::<Result<_, _>>()
         .context(ErrorKind::ConfigMissingKeys)?;
 
-    match string_vec[0].as_str()
-        {
-            "postgres" | "postgresql" => (),
-            _ => Err(ErrorKind::UnsupportedDbScheme)?,
-        }
-
-    Ok(format!(
-        "{scheme}://{username}:{password}@{host}:{port}/{database}",
-        scheme = string_vec[0],
-        username = string_vec[1],
-        password = string_vec[2],
-        host = string_vec[3],
-        port = string_vec[4],
-        database = string_vec[5],
-    ))
-}
+    let supported_schemes = ["postgres", "postgresql", "mysql", "sqlite"];
+    if !supported_schemes.contains(&string_vec[0].as_str()) {
+        Err(ErrorKind::UnsupportedDbScheme)?;
+    }
 
-#[derive(Debug, thiserror::Error)]
-#[error("Configuration was missing expected keys: [{:?}]", _0)]
-pub struct MissingKeys(Vec<String>);
-
-#[derive(Clone, Copy, Debug, Eq, thiserror::Error, Hash, PartialEq)]
-pub enum ErrorKind {
-    #[error("Unsupported database scheme, only 'postgres' and 'postgresql' are allowed.")]
-    UnsupportedDbScheme,
-    #[error("Configuration was missing expected keys")]
-    ConfigMissingKeys,
-    #[error("Config struct cannot be modified")]
-    ConfigImmutable,
+    Ok(concat!(
+        string_vec[0], "://", string_vec[1], ":", string_vec[2], "@", string_vec[3], ":", string_vec[4], "/", string_vec[5],
+    ).to_string())
 }
 
 #[cfg(feature = "simple-logging")]
-pub fn begin_log(config: &config::Config) {
-    use log::LevelFilter;
-
-    match config.get_string("Log.file").unwrap().as_ref() {
-        "_CONSOLE_" => (),
-        l => simple_logging::log_to_file(l, LevelFilter::Debug).unwrap(),
+/// Begins logging based on the provided config and log level
+pub fn begin_log(config: &config::Config, level: LevelFilter) -> Result<(), Box<dyn std::error::Error>> {
+    match config.get_string("Log.file")?.as_ref() {
+        "_CONSOLE_" => Ok(()),
+        l => {
+            simple_logging::log_to_file(l, level)?;
+            Ok(())
+        },
     }
```

Comment:
**Suggestion:** In the `begin_log` function, you are using a match statement to check if the log file is "_CONSOLE_". This can be simplified using an if statement.
```suggestion
    let log_file = config.get_string("Log.file")?;
    if log_file != "_CONSOLE_" {
    simple_logging::log_to_file(&log_file, level)?;
    }
    Ok(())
```